docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,842 @@
+
+### Removing dependencies
+
+With `SimpleIoc`, you would unregister your dependencies with the following method call.
+
+```csharp
+SimpleIoc.Default.Unregister<INavigationService>();
+```
+
+There is no direct replacement for removing dependencies with the MVVM Toolkit `Ioc` implementation.
+
+### Preferred constructor
+
+When registering your dependencies with MvvmLight's `SimpleIoc`, you have the option in your classes to provide a `PreferredConstructor` attribute for those with multiple constructors. 
+
+This attribute will need removing where used. 
+
+The dependency injection within the MVVM Toolkit's `Ioc` determines the best constructor based on the dependencies that have been added to the collection.
+
+The example below shows a dependency that relies on other dependencies with multiple constructors.
+
+```csharp
+public class NavigationService : INavigationService
+{
+    public NavigationService()
+    {
+    }
+
+    public NavigationService(IDialogService dialogService) : this(null, dialogService)
+    {
+    }
+
+    public NavigationService(IToastService toastService, IDialogService dialogService)
+    {
+    }
+}
+```
+
+With your dependencies registered as below, the `NavigationService(IToastService, IDialogService)` constructor will be called as both of the required dependencies are registered also.
+
+```csharp
+Ioc.Default.ConfigureServices(services =>
+{
+    services.AddSingleton<IDialogService, DialogService>();
+    services.AddSingleton<IToastService, ToastService>();
+    services.AddSingleton<INavigationService, NavigationService>();
+});
+```
+
+## Migrating Messenger
+
+The following steps focus on migrating your existing components which take advantage of the `Messenger` of the MvvmLight Toolkit. 
+
+The Windows Community Toolkit MVVM framework provides a [`Messenger`](Messenger) type that provides similar functionality. 
+
+A note on `Register` methods throughout this migration. MvvmLight uses weak references to establish the link between the `Messenger` instance and the recipient. This is not required by the MVVM Toolkit implementation and if this optional parameter has been set to `true` in any of your `Register` method calls, this will be removed.
+
+Below are a list of migrations that will need to be performed if being used in your current solution.
+
+The first change here will be swapping using directives in your components.
+
+```csharp
+// MvvmLight
+using GalaSoft.MvvmLight.Messaging;
+
+// Toolkit.Mvvm
+using Microsoft.Toolkit.Mvvm.Messaging;
+```
+
+### Messenger Methods
+
+#### Register<TMessage> ( object, Action<TMessage> )
+
+The functionality of `Register<TMessage>(object, Action<TMessage>)` can be achieved with the MVVM Toolkit's `IMessenger` extension method `Register<TMessage>(object, Action<TMessage>)`. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage>(this, this.OnMyMessageReceived);
```

Comment:
Assuming we're within `ObservableValidator` and can use `Messenger`, this would now be:

```suggestion
Messenger.Register<MyViewModel, MyMessage>(this, (r, m) => r.OnMyMessageReceived(m));
```

This is slightly more verbose but fixes all the issues with `MvvmLight` mentioned above. This approach gives us no captures, hence no allocations, and lets us properly use weak references behind the scenes without having to use reflection at all, and without having that weird `bool keepAlive` property from `MvvmLight` 😄